vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -731,8 +731,24 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
           }
           updateGridItemsInDomBasedOnCache(items);
         }
+        const cacheAndIndex = grid._cache.getCacheAndIndexByKey(pkey);
+        clearSubcache(cacheAndIndex.cache, cacheAndIndex.scaledIndex, index, length);
       });
 
+      /*
+       * Deletes items from grid cache, including the descendants of the affected
+       * items.
+       */
+      function clearSubcache(cache,scaledIndex, index, length) {
```

Comment:
Due to the comment below, you don't need to pass the (`cache, scaledIndex`) combo here at all, but just the `childCache`. If you're going to use this same function for covering the root cache also, you might want to rename the function (and the `childCache` parameter). Maybe `clearCacheRange(cache, startIndex, endIndex)`